src/ProgramBlockEditor.test.js

```diff
@@ -233,3 +240,77 @@ test('blocks', () => {
     expect(mockSelectHandler.mock.calls.length).toBe(6);
     expect(mockSelectHandler.mock.calls[5][0]).toBeNull();
 })
+
+
+test('Whenever active program step number updates, auto scroll to the step', () => {
+    const mockScrollInto = jest.fn();
+
+    window.HTMLElement.prototype.scrollIntoView = mockScrollInto;
+
+    const wrapper = mount(
+        <ProgramBlockEditor
+            activeProgramStepNum={0}
+            editingDisabled={true}
+            minVisibleSteps={6}
+            program={['forward', 'left', 'forward', 'left']}
+            selectedAction={null}
+            runButtonDisabled={false}
+            onClickRunButton={()=>{}}
+            onSelectAction={()=>{}}
+            onChange={()=>{}} />,
+        {
+            wrappingComponent: IntlProvider,
+            wrappingComponentProps: {
+                locale: 'en',
+                defaultLocale: 'en',
+                messages: messages.en
+            }
+        }
+    );
+
+    wrapper.setProps({ activeProgramStepNum: 1 });
+    expect(mockScrollInto.mock.calls.length).toBe(1);
+    expect(mockScrollInto.mock.calls[0][0]).toStrictEqual({ behavior: 'smooth', block: 'start', inline: 'start' });
+
+    wrapper.setProps({ activeProgramStepNum: 2});
+
+    expect(mockScrollInto.mock.calls.length).toBe(2);
+    expect(mockScrollInto.mock.calls[1][0]).toStrictEqual({ behavior: 'smooth', block: 'start', inline: 'start'});
+
+});
+
+test('When the program is running (if activeProgramStepNum prop has a value), editor action buttons should be disabled', () => {
```

Comment:
The logic to disable no longer depends on activeProgramStepNum -- update the test title:

'The editor action buttons disabled states are set according to the editingDisabled property'